app/controllers/api/v1/vacation_requests_controller.rb

```diff
@@ -9,14 +15,55 @@ def create
     end
   end
 
+  def update
+    vacation_request = VacationRequest.find params[:id]
+
+    vacation_request.attributes = vacation_request_params
+
+    if vacation_request.save
+      render json: vacation_request, status: :ok
+    else
+      render json: vacation_request.errors, status: :unprocessable_entity
+    end
+  end
+
+  def approve
+    render json: false, status: 503 and return unless current_user.admin
+
+    vacation_request = VacationRequest.find params[:id]
+    vacation_request.approve!
+
+    render json: vacation_request, status: :ok
+  end
+
+  def reject
+    render json: false, status: 503 and return unless current_user.admin
+
+    vacation_request = VacationRequest.find params[:id]
+    vacation_request.reject!
+
+    render json: vacation_request, status: :ok
+  end
+
   private
 
   def vacation_request_params
```

Comment:
nice! I didnt think of this way of separating the params for different users :dancer: